daemon/src/db.rs

```diff
@@ -19,15 +22,37 @@ use sqlx::pool::PoolConnection;
 use sqlx::sqlite::SqliteConnectOptions;
 use sqlx::Sqlite;
 use sqlx::SqlitePool;
+use std::any::Any;
+use std::any::TypeId;
 use std::path::PathBuf;
+use std::sync::Arc;
 use time::Duration;
 
+#[derive(Clone)]
+pub struct Connection {
```

Comment:
love this wrapper, much less awkward than passing mut-ref to `Connections` to functions!